src/dmd/ctorflow.d

```diff
@@ -105,53 +105,118 @@ struct CtorFlow
 
 
 /****************************************
- * Merge fi flow analysis results into fieldInit.
+ * Merge `b` flow analysis results into `a`.
  * Params:
- *      fieldInit = the path to merge fi into
- *      fi = the other path
+ *      a = the path to merge `b` into
+ *      b = the other path
  * Returns:
- *      false means either fieldInit or fi skips initialization
+ *      false means one of the paths skips construction
  */
-bool mergeFieldInitX(ref CSX fieldInit, const CSX fi)
+bool mergeCallSuper(ref CSX a, const CSX b) pure nothrow
 {
-    if (fi != fieldInit)
+    // This does a primitive flow analysis to support the restrictions
+    // regarding when and how constructors can appear.
+    // It merges the results of two paths.
+    // The two paths are `a` and `b`; the result is merged into `a`.
+    bool ok = true;
+    if (b != a)
```

Comment:
The function does nothing and returns `true` if `a == b`, yet it takes the reader a whopping 47 lines to figure that out. Can we please please please with Truvia on top renounce this crap:

```d
bool ok = true;
if (condition)
{
    ... so many lines ...
}
return ok;
```

This antipattern is all over the compiler code. It's insufferable in old code, and definitely not tolerable in the sligthest in new code. We're not coding Fortran over here. Please:

```d
if (!condition)
{
    return true;
}
... knock yourself out ...
```